smarts/sstudio/types.py

```diff
@@ -852,6 +853,55 @@ def to_geometry(self, road_map: Optional[RoadMap] = None) -> Polygon:
         return shapely_translate(poly, xoff=x, yoff=y)
 
 
+@dataclass(frozen=True)
+class ConfigurableZone(Zone):
+    """A descriptor for a zone with user-defined geometry."""
+
+    ext_coordinates: List[Tuple[float, float]]
+    """external coordinates of the polygon
+    < 2 points provided: error
+    = 2 points provided: generates a box using these two points as diagonal
+    > 2 points provided: generates a polygon according to the coordinates"""
+    rotation: Optional[float] = None
+    """The heading direction of the bubble(radians, clock-wise rotation)"""
+
+    def __post_init__(self):
+        if (
+            not self.ext_coordinates
+            or not isinstance(self.ext_coordinates[0], tuple)
+            or len(self.ext_coordinates) < 2
```

Comment:
We should swap these conditions to short circuit and prevent an index error.
```suggestion
            or len(self.ext_coordinates) < 2
            or not isinstance(self.ext_coordinates[0], tuple)
```